src/Themes/ui/ColorPicker/index.tsx

```diff
@@ -0,0 +1,246 @@
+}
+
+export function OpenColorPickerButton({ color, onColorChange }: OpenColorPickerButtonProps) {
+  const [open, setOpen] = useState(false);
+
+  //detached state to prevent rerenders of modal on change.
+  const [selectedColor, setSelectedColor] = useDetachedState("");
+
+  return (
+    <button
+      style={{
+        position: "relative",
+        margin: "6px",
+        width: "24px",
+        minWidth: "24px",
+        height: "24px",
+        minHeight: "24px",
+        borderRadius: "5px",
+        border: "none",
+        cursor: "pointer",
+        backgroundColor: color,
+        color: "inherit",
+      }}
+      onClick={() => open || setOpen(true)}
+    >
+      <Modal open={open} onClose={() => setOpen(false)}>
+        {open && (
+          <>
+            <ColorPicker setColorString={(c) => setSelectedColor(c)} initialColor={parseHEXtoRGB(color)}></ColorPicker>
+            <div
+              style={{
+                display: "flex",
+                justifyContent: "right",
+                marginTop: "1em",
+              }}
+            >
+              <Button onClick={() => onColorChange(selectedColor[0])}>Apply</Button>
+            </div>
+          </>
+        )}
+      </Modal>
+    </button>
+  );
+}
+
+type ColorPickerProps = {
+  initialColor?: RGB | HSV;
+  setColorString: (color: string) => void;
+} & React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>;
+
+export function ColorPicker({ initialColor, setColorString, ...attr }: ColorPickerProps) {
+  const [color, setColor] = useState<RGB | HSV>(initialColor ?? { r: 255, g: 255, b: 255 });
+
+  const { rgb, hsv } = normalize(color);
+
+  const theme = useTheme();
+
+  useEffect(() => {
+    setColorString(formatRGBtoHEX(normalize(color).rgb));
+  }, [color, setColorString]);
+
+  return (
+    <div
+      {...attr}
+      style={{
+        height: "250px",
+        width: "350px",
+        display: "flex",
+        flexDirection: "column",
+        userSelect: "none",
+        gap: "1em",
+        paddingTop: "8px",
+        color: theme.palette.primary.light,
+        fontFamily: theme.typography.fontFamily,
+      }}
+    >
+      <SVCanvas hue={hsv.hue} sat={hsv.sat} val={hsv.val} setSV={(sv) => setColor({ hue: hsv.hue, ...sv })}></SVCanvas>
+      <HueSlider hue={hsv.hue} setHue={(h) => setColor({ ...hsv, hue: h })}></HueSlider>
+      <span
+        style={{
+          display: "flex",
+          justifyContent: "space-between",
+        }}
+      >
+        <ColorPreview rgb={rgb}></ColorPreview>
+        <HexInput rgb={rgb} setRGB={(c) => setColor(c)}></HexInput>
+        <RGBDigit digitLabel="R" digit={[rgb.r, (r) => setColor({ ...rgb, r })]}></RGBDigit>
+        <RGBDigit digitLabel="G" digit={[rgb.g, (g) => setColor({ ...rgb, g })]}></RGBDigit>
+        <RGBDigit digitLabel="B" digit={[rgb.b, (b) => setColor({ ...rgb, b })]}></RGBDigit>
+      </span>
+    </div>
+  );
+}
+
+function useDetachedState<T>(val: T) {
```

Comment:
I see what you are trying to do here with useDeferredState. But, it is a bad idea, and what you *actually* want to be doing is covered by other things that are builtin to React.

Apologies if I go over stuff you already know, but I want to be thorough here. Core to understanding what needs to happen here is understanding how React works, and then also understanding how React *doesn't* work, but can be made to handle.

Normally React operates by dealing with *state*. Every time state changes, React will rerender *everything* in that component, which recursively means all components below it. However, rerendering *doesn't* (necesarily) mean destroying and recreating DOM nodes, so typically this is pretty fast and doesn't cause any visible artifacts.

However, that approach doesn't work well for stuff like entering text in a textbox, or moving a slider. If we managed that with React state, then *every single character* would cause a rerender of the textbox, and every tiny nudge of the mouse would cause a rerender of the slider. The performance doesn't hold up.

For these scenarios, we fall back on native DOM functionality: we tell React to get the hell out of the way, we'll let the browser do its thing thank you. React calls this distinction "controlled vs uncontrolled" components: A controlled component is controlled with a state variable, and is always locked to the value of that state (via rerenders), while an uncontrolled component is not (the DOM will do its thing). 

In your case, you are definitely trying to construct your overall component as an uncontrolled component, but you need to be using different approaches to do it. At the center of everything is the `useRef` method, which creates a *mutable reference* that lives with your component, and does not create rerenders. You can use the `ref` field in JSX to pass a ref object in, and it will get assigned the *DOM object* (as opposed to the React component), which lets you mutate the DOM in handlers. This is what you are *actually* trying to do in your mousedown on the SVCanvas, for instance.

I imagine when this is done you will mostly have various refs to specific DOM objects (the gradient, the textboxes) and very little actual state. `setDetachedState` and `setSyncState` will be gone.

See these pages for more details:
<https://react.dev/learn/manipulating-the-dom-with-refs>
<https://legacy.reactjs.org/docs/uncontrolled-components.html>
<https://react.dev/reference/react-dom/components/input>